src/yunohost/hook.py

```diff
@@ -311,14 +313,56 @@ def hook_exec(path, args=None, raise_on_error=False, no_trace=False,
         user -- User with which to run the command
 
     """
-    from moulinette.utils.process import call_async_output
 
     # Validate hook path
     if path[0] != '/':
         path = os.path.realpath(path)
     if not os.path.isfile(path):
         raise YunohostError('file_does_not_exist', path=path)
 
+    # Check the type of the hook (bash by default)
+    hook_type = "bash"
+    # (non-bash hooks shall start with something like "#!/usr/bin/env language")
+    hook_ext = os.path.splitext(path)[1]
```

Comment:
https://gist.github.com/LeMeteore/cac019562822388a5d11c0ead629848c#file-transform-py-L35-L37 you might want to use this kind of code for file type detection (that also handle python script without a ".py" at the end but a shebang)